includes/class-sensei-analysis-overview-list-table.php

```diff
@@ -1043,41 +1043,34 @@ public function filter_courses_by_last_activity( array $clauses ): array {
 		if ( ! $start_date && ! $end_date ) {
 			return $clauses;
 		}
-
-		// Join the lessons meta.
-		// We need the lesson ids to extract the last activity comment.
-		$clauses['join'] .= "
-			INNER JOIN {$wpdb->postmeta} AS pm ON pm.meta_value = {$wpdb->posts}.ID
-			AND pm.meta_key = '_lesson_course'
-		";
-
-		// Join only the last activity comment.
-		// Following the logic from `Sensei_Analysis_Overview_List_Table::get_last_activity_date()`.
-		$clauses['join'] .= " INNER JOIN {$wpdb->comments} AS c ON c.comment_ID = (
-			SELECT comment_ID
-			FROM {$wpdb->comments}
-			WHERE {$wpdb->comments}.comment_post_ID = pm.post_id
-			AND {$wpdb->comments}.comment_approved IN ('complete', 'passed', 'graded')
-			AND {$wpdb->comments}.comment_type = 'sensei_lesson_status'
-			ORDER BY {$wpdb->comments}.comment_date_gmt DESC
-			LIMIT 1
-		)";
-
+		// Fetch the lessons within the expected last activity range.
+		$lessons_query = "SELECT cm.comment_post_id lesson_id, MAX(cm.comment_date_gmt) as comment_date_gmt
```

Comment:
I'm not sure if `MAX(cm.comment_date_gmt) as comment_date_gmt` is actually used in this query. Could it be removed?

Reply:
It is used only in the case of end_date filtering. First we are taking the max for each lesson and kind of creating a second table, then from those maxes, we are choosing the max for the entire course and filtering based on that. This end_date part is a little tricky :p I got confused a couple of times while writing it.